client/src/app/Pages/user-messages/user-messages.component.ts

```diff
@@ -0,0 +1,96 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { UserService } from '../../Services/user.service';
+
+@Component({
+  selector: 'app-user-messages',
+  templateUrl: './user-messages.component.html',
+  styleUrls: ['./user-messages.component.css'],
+})
+export class UserMessagesComponent implements OnInit {
+  messages: any[] = [];
+  currentPage: number = 1;
+  pageSize: number = 10;
+  hasMoreData: boolean = true;
+  messageResult: string | null = null;
+
+  messageForm: FormGroup;
+
+  allMessageFields: string[] = [
+    'MessageId',
+    'ChatId',
+    'UserId',
+    'MessageText',
+    'SentAt',
+    'SenderUserName',
+  ];
+  displayedFields: string[] = this.allMessageFields;
+
+  dropdownList: string[] = this.allMessageFields;
+  selectedItems: string[] = this.allMessageFields;
+  dropdownSettings = {};
+
+  constructor(private userService: UserService, private fb: FormBuilder) {
+    this.messageForm = this.fb.group({
+      sortBy: ['SentAt', Validators.required],
+      searchValue: [''],
+      fields: [this.selectedItems],
+      ascending: [true],
+    });
+  }
+
+  ngOnInit(): void {
+    this.dropdownSettings = {
+      singleSelection: false,
+      idField: 'item',
+      textField: 'item',
+      selectAllText: 'Select All',
+      unSelectAllText: 'UnSelect All',
+      allowSearchFilter: true,
+    };
+    this.loadMessages();
+  }
+
+  loadMessages(page: number = 1): void {
```

Comment:
This method should be split to different ones

Reply:
This will be fixed in next PRs